File: azan_local_notifications/android/src/main/java/com/ahmed/azan_local_notifications/models/styles/MessagingStyleInformation.java
package com.ahmed.azan_local_notifications.models.styles;

import androidx.annotation.Keep;

import com.ahmed.azan_local_notifications.models.MessageDetails;
import com.ahmed.azan_local_notifications.models.PersonDetails;

import java.util.ArrayList;

@Keep
public class MessagingStyleInformation extends DefaultStyleInformation {
  public PersonDetails person;
  public String conversationTitle;
  public Boolean groupConversation;
  public ArrayList<MessageDetails> messages;

  public MessagingStyleInformation(
      PersonDetails person,
      String conversationTitle,
      Boolean groupConversation,
      ArrayList<MessageDetails> messages,
      Boolean htmlFormatTitle,
      Boolean htmlFormatBody) {
    super(htmlFormatTitle, htmlFormatBody);
    this.person = person;
    this.conversationTitle = conversationTitle;
    this.groupConversation = groupConversation;
    this.messages = messages;
  }
}

File: azan_local_notifications/android/src/main/kotlin/com/ahmed/azannotifications/models/styles/MessagingStyleInformation.kt
package com.ahmed.azanlocalnotifications.models.styles

import androidx.annotation.Keep
import com.dexterous.flutterlocalnotifications.models.MessageDetails
import com.dexterous.flutterlocalnotifications.models.PersonDetails
import java.util.ArrayList

@Keep
class MessagingStyleInformation(
    person: PersonDetails,
    conversationTitle: String,
    groupConversation: Boolean,
    messages: ArrayList<MessageDetails?>,
    htmlFormatTitle: Boolean?,
    htmlFormatBody: Boolean?
) : DefaultStyleInformation(htmlFormatTitle, htmlFormatBody) {
    var person: PersonDetails
    var conversationTitle: String
    var groupConversation: Boolean
    var messages: ArrayList<MessageDetails>

    init {
        this.person = person
        this.conversationTitle = conversationTitle
        this.groupConversation = groupConversation
        this.messages = messages
    }
}